azuredevops/client_repository_pullrequest.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2023 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azuredevops
+
+import (
+	"context"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/git"
+)
+
+// PullRequestClient implements the gitprovider.PullRequestClient interface.
+var _ gitprovider.PullRequestClient = &PullRequestClient{}
+
+// PullRequestClient operates on the pull requests for a specific repository.
+type PullRequestClient struct {
+	*clientContext
+	ref gitprovider.RepositoryRef
+}
+
+// List lists all pull requests for the repository.
+func (c *PullRequestClient) List(ctx context.Context) ([]gitprovider.PullRequest, error) {
+	repositoryId := c.ref.GetRepository()
+	projectId := c.ref.GetIdentity()
+	prOpts := git.GetPullRequestsArgs{
+		RepositoryId:   &repositoryId,
+		Project:        &projectId,
+		SearchCriteria: &git.GitPullRequestSearchCriteria{},
+	}
+	prs, err := c.g.GetPullRequests(ctx, prOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	requests := make([]gitprovider.PullRequest, len(*prs))
+
+	for idx, pr := range *prs {
+		requests[idx] = newPullRequest(c.clientContext, &pr)
```

Comment:
```suggestion
		pr := pr
		requests[idx] = newPullRequest(c.clientContext, &pr)
```

This is needed because Go reuses variables. You may end up with `requests` pointing to the same address for all items in the slice.